### ISSUE TITLE
[Review] Optional email notifications

### ISSUE THREAD

**Issue by JustusAdam** [fsr/lesson-builder#4]
Criticism has been raised, because failed builds are not properly reported, except for being dumped into the log.

Therefore I ask the people who actually maintain the lessons, **would you be interested in receiving (optional) email notifications upon a failed build?**

Secondly: If yes, then should the emails be scoped by repository? Aka maintainer emails for each watched repository or global?

Local would look something like this in the config (`watch_conf.json`)

``` json
{
    "watched": [
        {
            "name": "fsr/python-lessons",
            "directory": "python_lessons",
            "maintainers": "max.musterman@gmail.com",
            "notify-on-fail": true
        }
    ]
}
```

Or global:

``` json
{
    "watched": [
        {
            "name": "fsr/python-lessons",
            "directory": "python_lessons"
        }
    ],
    "maintainers": "max.musterman@gmail.com",
    "notify-on-fail": true
}
```

If `maintainers` contains an array with multiple addresses all of them would get notified.

I am also open for suggestions as to the content of the email. Should it contain relevant parts of the log already or just say something like "the build failed, refer to the log for information".

This might me interesting to @Feliix42, @h4llow3En, @manuelthieme. Please feel free to tag other people for whom this could be of interest.

**Comment by Feliix42** [comment]
I think it would be good practice to only notify the person who pushed the commit.
But I also see the advantages of notifying a group of maintainers.

For the content I think a simple failure message and a path/link to the log would be enough as LaTeX errors can be rather confusing and hard to parse via script.

Other suggestions?

**Comment by JustusAdam** [comment]
actually I could perhaps query git ...

**Comment by Feliix42** [comment]
I see. And using the mail address one has to supply to `git` might not be the best solution. And when someone pushes a commit via PR he would get the mail, which wouldn't be helpful either.
So then I stick with the local configuration per repository.

**Comment by JustusAdam** [comment]
You could also just write bogus for your email address in git, even though that is not recommendable. And it puts the burden to fix the error on the submitter, who might not necessarily rely on the build succeeding and maybe doesn't care. It should go to those people who need the files being built so that they know they are out of date.

**Comment by Feliix42** [comment]
I agree 100% with you.
